grype/matcher/rpm/almalinux_test.go

```diff
@@ -0,0 +1,1689 @@
+						},
+					},
+				},
+			},
+			unaffectedResults: result.Set{
+				"ALSA-2023:1234": []result.Result{
+					{
+						ID: "ALSA-2023:1234",
+						Vulnerabilities: []vulnerability.Vulnerability{
+							{
+								Reference: vulnerability.Reference{ID: "ALSA-2023:1234"},
+								RelatedVulnerabilities: []vulnerability.Reference{
+									{ID: "CVE-2023-1234"}, // ALSA has CVE as alias
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedVulnCount: 1,
+			expectedVulnIDs:   []string{"CVE-2023-5678"}, // CVE-2023-1234 should be filtered out
+		},
+		{
+			name: "CVE disclosure filters out ALSA by alias",
+			disclosures: result.Set{
+				"ALSA-2023:1234": []result.Result{
+					{
+						ID: "ALSA-2023:1234",
+						Vulnerabilities: []vulnerability.Vulnerability{
+							{
+								Reference: vulnerability.Reference{ID: "ALSA-2023:1234"},
+								RelatedVulnerabilities: []vulnerability.Reference{
+									{ID: "CVE-2023-1234"},
+								},
+							},
+						},
+					},
+				},
+				"CVE-2023-5678": []result.Result{
+					{
+						ID: "CVE-2023-5678",
+						Vulnerabilities: []vulnerability.Vulnerability{
+							{Reference: vulnerability.Reference{ID: "CVE-2023-5678"}},
+						},
+					},
+				},
+			},
+			unaffectedResults: result.Set{
+				"CVE-2023-1234": []result.Result{
+					{
+						ID: "CVE-2023-1234",
+						Vulnerabilities: []vulnerability.Vulnerability{
+							{Reference: vulnerability.Reference{ID: "CVE-2023-1234"}},
+						},
+					},
+				},
+			},
+			expectedVulnCount: 1,
+			expectedVulnIDs:   []string{"CVE-2023-5678"}, // ALSA-2023:1234 should be filtered out by alias
+		},
+		{
+			name: "multiple aliases - complex filtering",
+			disclosures: result.Set{
+				"CVE-2023-1234": []result.Result{
+					{
+						ID: "CVE-2023-1234",
+						Vulnerabilities: []vulnerability.Vulnerability{
+							{
+								Reference: vulnerability.Reference{ID: "CVE-2023-1234"},
+								RelatedVulnerabilities: []vulnerability.Reference{
+									{ID: "GHSA-abcd-1234"},
+								},
+							},
+						},
+					},
+				},
+				"CVE-2023-5678": []result.Result{
+					{
+						ID: "CVE-2023-5678",
+						Vulnerabilities: []vulnerability.Vulnerability{
+							{Reference: vulnerability.Reference{ID: "CVE-2023-5678"}},
+						},
+					},
+				},
+			},
+			unaffectedResults: result.Set{
+				"ALSA-2023:1234": []result.Result{
+					{
+						ID: "ALSA-2023:1234",
+						Vulnerabilities: []vulnerability.Vulnerability{
+							{
+								Reference: vulnerability.Reference{ID: "ALSA-2023:1234"},
+								RelatedVulnerabilities: []vulnerability.Reference{
+									{ID: "GHSA-abcd-1234"}, // matches alias of CVE-2023-1234
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedVulnCount: 1,
+			expectedVulnIDs:   []string{"CVE-2023-5678"}, // CVE-2023-1234 filtered by alias match
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			filtered := tt.disclosures.Remove(tt.unaffectedResults)
```

Comment:
I think this is already covered by https://github.com/anchore/grype/blob/v0.101.0/grype/matcher/internal/result/results_test.go#L118-L285 in the result package